dwave/gate/simulator/operation_generation.py

```diff
@@ -28,23 +28,34 @@ def binary(i: int, n: int) -> Tuple[int, ...]:
 
 
 def compile_gate(
-    gate_matrix: Union[np.ndarray, List[List[EntryType]]]
+    gate_matrix: Union[np.ndarray, List[List[EntryType]]],
+    skip_identity: bool = True,
+    skip_zeros: bool = False,
 ) -> Tuple[Set[int], Dict[int, List[Tuple[int, EntryType]]]]:
     """Compile a given gate into a minimal set of instructions.
 
     Args:
-        gate_matrix: Square matrix representing the gate. Values may be scalars or C variables that
-            refer to floats.
+        gate_matrix: Square matrix representing the gate. Values may be scalars or C
+            variables that refer to floats.
+
+        skip_identity: If true, rows of given matrix that are equivalent to the identity
+            operation are ignored.
+
+        skip_zeros: If true, rows of given matrix that have no values will be ignored.
 
     Returns:
-        A tuple where the first item is the set of substates acted on by the gate, and the second
-        item is a dictionary mapping output substates to those corresponding instructions.
+        A tuple where the first item is the set of substates acted on by the gate, and
+        the second item is a dictionary mapping output substates to those corresponding
+        instructions.
```

Comment:
Should the `output_sub_states` set be added to the description as well? Also, would it be more correct to say that the the dictionary is "mapping _dependent_ substates to those corresponding instructions", based on what they're named in the code?

Reply:
Yes, good catch! Fixed